spring-xd-messagebus-kafka/src/main/java/org/springframework/xd/dirt/integration/kafka/KafkaMessageBus.java

```diff
@@ -789,4 +806,43 @@ private int roundRobin() {
 
 	}
 
+	@Override
+	public void storeForManualAck(MessageHeaders headers) {
+		String topic = (String) headers.get(KafkaHeaders.TOPIC);
+		Partition partition = new Partition(topic, (int) headers.get(KafkaHeaders.PARTITION_ID));
+		Long offset = (Long) headers.get(KafkaHeaders.OFFSET);
```

Comment:
Unfortunately, the proper header here should be "nextOffset", which is missing from the current implementation of SI-Kafka. The known issue here is that if we store the current offset, then in case of a crash, reading will replay the last item. This is addressed as part of https://github.com/spring-projects/spring-integration-kafka/pull/20